### PR TITLE
Integrate amplitude

### LINKED ISSUE #180: Integrate telemetry
Related to this spec: https://github.com/meilisearch/specifications/pull/34/files
Replaces #141 

----

Integrate telemetry (Amplitude) in the same way this is already implemented in MeiliSearch v0.20.0 except:
- [x] we need to use [`http-api-v2` of Amplitude](https://developers.amplitude.com/docs/http-api-v2). The current version of `http-api` in MeiliSearch is deprecated.
- [x] the Sentry and Amplitude errors should fail silently
- [x] the flag `--no-sentry` and the variable `MEILI_NO_SENTRY` are removed. Now, the flag `--no-analytics` (env vraiable `MEILI_NO_ANALYTICS`) disables Amplitude AND Sentry at the same time.
- [x] When launching MeiliSearch by default, the telemetry is still enabled but the following message should be present (decided in the spec above):
```text
...
Commit SHA:		"def1596eaf95c56a964399c4e5ae0875e0b155ee"
Commit date:		"2021-06-15"
Package version:	"0.21.0-alpha.4"

Thank you for using MeiliSearch!

We collect anonymized analytics to improve our product and your experience. To learn more, including how to turn off analytics, visit our dedicated documentation page: https://docs.meilisearch.com/reference/features/configuration.html#analytics

Anonymous telemetry: "Enabled"

Documentation:		https://docs.meilisearch.com
Source code:		https://github.com/meilisearch/meilisearch
...
```

- [X] If the telemetry is disabled, here is the following message 
```text
...
Commit SHA:		"def1596eaf95c56a964399c4e5ae0875e0b155ee"
Commit date:		"2021-06-15"
Package version:	"0.21.0-alpha.4"
Anonymous telemetry:    "Disabled"

Documentation:		https://docs.meilisearch.com
Source code:		https://github.com/meilisearch/meilisearch
...
```

- [X]  Remove the possibility to choose the sentry server

I put the `breaking` label for the docs team (@react-learner) because:
- the message will change when starting MeiliSearch, so the screenshot might be obsolete 🙂 
- the flag `--no-sentry` and the variable `MEILI_NO_SENTRY` are removed. Cf third point of this issue (related to #204)

- [X] Deactivate the collection of analytics in debug build

### PR DESCRIPTION
**Pull request by irevoire**
And merge the sentry and amplitude usage under one “Enable analytics” flag

closes #180

### CODE REVIEW

**Review by curquiza — CHANGES_REQUESTED** [review]

File: meilisearch-http/src/main.rs [hunk 1]
```diff
@@ -50,25 +55,14 @@ async fn main() -> Result<(), MainError> {
...
-
-    //if opt.schedule_snapshot {
-    //snapshot::schedule_snapshot(data.clone(), &opt.snapshot_dir, opt.snapshot_interval_sec.unwrap_or(86400))?;
-    //}
+    #[cfg(feature = "analytics")]
+    if !opt.no_analytics {
+        let analytics_data = data.clone();
+        let analytics_opt = opt.clone();
+        tokio::task::spawn(analytics::analytics_sender(analytics_data, analytics_opt));
+    }
```
Comment (MarinPostma) [inline meilisearch-http/src/main.rs:65]: Maybe we should prevent data to be sent in debug builds, just like above

**Review by MarinPostma — CHANGES_REQUESTED** [review]

File: meilisearch-http/src/main.rs (hunk 1 above)
Comment (irevoire) [reply meilisearch-http/src/main.rs:65]: This was not the case in the current Meilisearch. But if @qdequele or @gmourier are ok with that, I can do it :slightly_smiling_face:

**Review by curquiza — DISMISSED** [review]

File: meilisearch-http/src/main.rs (hunk 1 above)
Comment (gmourier) [reply meilisearch-http/src/main.rs:65]: To be sure, are we talking about a MeiliSearch compiled by the user in debug mode and not in release mode? I ask the question to be sure that it does not concern the log level. 

If this is the case, I don't see any immediate need to have analytics on the uses of Meilisearch when it is compiled in debug. It also avoids polluting Amplitude with data that doesn't represent the majority of our users.

Comment (irevoire) [reply meilisearch-http/src/main.rs:65]: Yes, you understood perfectly. Right now Meilisearch (and this pr) are sending amplitude analytics even when compiled in Debug mode. Sentry is not sending any analytics while in debug mode though.

Basically, the debug mode is faster to compile, slower to execute and obviously easier to debug.
Every binary we ship is compiled in release mode so someone using a debug build build it himself from the source.

Comment (gmourier) [reply meilisearch-http/src/main.rs:65]: The spec has been updated to mention the debug build and state we don't collect any telemetry in debug build!

**Comment by irevoire** [comment]
I removed the usage of amplitude when in debug mode :slightly_smiling_face:

**Review by MarinPostma — APPROVED** [review]
bors merge

**Pull request merged**